vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java

```diff
@@ -1005,7 +969,18 @@ private String generateHtmlValue() {
          */
         @Override
         public void setValue(String value) {
-            RichTextEditor.this.setValue(value);
+            RichTextEditor.this.getElement().setProperty("value", value);
+            // After setting delta value, manually sync back the updated HTML
+            // value, which will eventually trigger a server-side value change
+            // event on the component
+            RichTextEditor.this.getElement().executeJs("return this.htmlValue")
+                    .then(jsonValue -> {
+                        isHtmlValueSync = true;
+                        RichTextEditor.this.setValue(jsonValue.asString());
+                        isHtmlValueSync = false;
+                    });
+            fireChangeEvent(false);
+            oldValue = value;
```

Comment:
And this is the reverse sync. When setting the delta value, we set the `value` property, and then retrieve the updated `htmlValue`. Using that we update the component's value.

Note that this is significantly easier than the sync logic above, because we can use `executeJs`. It seems that when setting `value`, then `htmlValue` is immediately updated in the web component. However calling `dangerouslySetHtmlValue` does not immediately update `value`, thus the workaround with the value change listener above.